File: OOP/6/poly/bin/shape.dart
class Shape {
  // artribut
  double luas;
  double keliling;

  // method
  double hitungLuas() => 1;
  double hitungKeliling() => 1;
}

class Circle extends Shape {
  double r;

  Circle(this.r);

  @override
  double hitungLuas() => 3.14 * r * r;

  @override
  double hitungKeliling() => 3.14 * r * 2;
}

class Polygon extends Shape {
  int jumlahSisi;
}

class Triangle extends Polygon {
  double alas;
  double tinggi;

  Triangle(this.alas, this.tinggi);

  @override
  double hitungLuas() => alas * tinggi / 2;

  @override
  double hitungKeliling() => alas * 3;
}

class Rectangle extends Polygon {
  double panjang;
  double lebar;

  Rectangle(this.panjang, this.lebar);

  @override
  double hitungLuas() => panjang * lebar;

  @override
  double hitungKeliling() => panjang * 2 + lebar * 2;
}

File: week3/prak6/Shape.java
class Shape{
	public double length;
	public double width;
	public double height;
	public double volume;

	public Shape(double length, double width, double height){
		this.length = length;
		this.width = width;
		this.height = height;
		calculateVolume();
	}

	public double compareTo(Shape shape){
		return shape.volume - this.volume;
	}

	private void calculateVolume(){
		this.volume = this.length * this.width * this.height;
	}

	public void cetak(){
		System.out.printf("Panjang x lebar x tinggi = %.0f x %.0f x %.0f\n", this.length, this.width, this.height);
		System.out.printf("Volume = %.0f\n", this.volume);
	}
}
